lending_pool/tests/rust_tests.rs

```diff
@@ -336,3 +338,349 @@ fn repay_test() {
+            |sc| {
+                sc.deposit_asset(managed_address!(&borrower_addr));
+            },
+        )
+        .assert_ok();
+
+    // Check LEGLD
+    lending_setup.b_mock.check_nft_balance(
+        &borrower_addr,
+        LEND_EGLD,
+        1,
+        &rust_biguint!(1_000),
+        Option::<&Empty>::None,
+    );
+
+    // Borrow USDC
+    lending_setup
+        .b_mock
+        .execute_esdt_transfer(
+            &borrower_addr,
+            &lending_setup.liquidity_pool_usdc_wrapper,
+            LEND_EGLD,
+            1,
+            &rust_biguint!(1_000),
+            |sc| {
+                sc.borrow(
+                    managed_address!(&borrower_addr),
+                    managed_biguint!(500_000_000),
+                );
+            },
+        )
+        .assert_ok();
+
+    // lending_setup.b_mock.dump_state_for_account_hex_attributes(&borrower_addr);
+
+    // Check received BUSDC
+    lending_setup.b_mock.check_nft_balance(
+        &borrower_addr,
+        BORROW_USDC_TOKEN_ID,
+        1,
+        &rust_biguint!(100_000),
+        Some(&Vec::<u8>::new()),
+    );
+
+    // Check received USDC
+    lending_setup.b_mock.check_nft_balance(
+        &borrower_addr,
+        USDC_TOKEN_ID,
+        0,
+        &rust_biguint!(100_000),
+        Some(&Vec::<u8>::new()),
+    );
+
+    lending_setup
+        .b_mock
+        .execute_tx(
+            &liquidator_addr,
+            &lending_setup.price_aggregator_wrapper,
+            &rust_biguint!(0),
+            |sc| {
+                sc.set_latest_price_feed(
+                    managed_buffer!(EGLD_TICKER),
+                    managed_buffer!(DOLLAR_TICKER),
+                    managed_biguint!(EGLD_PRICE_DROPPED_IN_DOLLARS),
+                );
+            },
+        )
+        .assert_ok();
+
+    lending_setup
+        .b_mock
+        .set_esdt_balance(&liquidator_addr, USDC_TOKEN_ID, &rust_biguint!(100_000));
+
+    lending_setup
+        .b_mock
+        .execute_esdt_transfer(
+            &liquidator_addr,
+            &lending_setup.liquidity_pool_usdc_wrapper,
+            USDC_TOKEN_ID,
+            0,
+            &rust_biguint!(100_000),
+            |sc| {
+                sc.liquidate(
+                    managed_address!(&liquidator_addr),
+                    1,
+                    managed_biguint!(5_000_000),
+                );
+            },
+        )
+        .assert_ok();
+
+    // Check Liquidator Balance - LEND_EGLD tokens
+    lending_setup.b_mock.check_nft_balance(
+        &liquidator_addr,
+        LEND_EGLD,
+        1,
+        &rust_biguint!(717),
+        Option::<&Empty>::None,
+    );
+
+    lending_setup
+        .b_mock
+        .execute_esdt_transfer(
+            &liquidator_addr,
+            &lending_setup.liquidity_pool_egld_wrapper,
+            LEND_EGLD,
+            1,
+            &rust_biguint!(717),
+            |sc| {
+                sc.reduce_position_after_liquidation();
```

Comment:
I'm not sure if assuming the liquidator will do this out of their own kindness after each liquidation is a good idea. I think this should somehow be an atomic operation, done through the lending pool:
- user calls liquidate on the lending pool
- the lending pool performs the liquidation on the relevant pool
- the lending pool then performs the reduce_position action on the other pool

Reply:
Yep, you are right. After this, the liquidator loses his funds :))
I wanted to also test the `reduce_position_after_liquidation` function.
For the final setup, the scenario is as you described.